src/app/shared/guards/current-user.guard.ts

```diff
@@ -0,0 +1,25 @@
+import { Injectable } from '@angular/core';
+import { CanActivate, ActivatedRouteSnapshot, RouterStateSnapshot, UrlTree, Router } from '@angular/router';
+import { Observable } from 'rxjs';
+import { AuthService } from '@service/auth.service';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class CurrentUserGuard implements CanActivate {
```

Comment:
Same as this [comment](https://github.com/carryall/survey-challenge/pull/29/files#r540744220)